doc/sphinx/language/extensions/evars.rst

```diff
@@ -74,18 +74,51 @@ Expressions often contain redundant pieces of information. Subterms that can be
 automatically inferred by Coq can be replaced by the symbol ``_`` and Coq will
 guess the missing piece of information.
 
+Automatic resolution of existential variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some tactics automatically resolve goals that are created when existential
+variables are created.
+
+.. _automatic-evar-resolution:
+
+.. example:: Automatic resolution of existential variables
+
+   .. coqtop:: reset in
+
+      Goal forall (a : nat), a = a -> exists n : nat, n = n.
+      Proof.
+        intros a e.
+
+   .. coqtop:: all
+
+       eexists ?[x].  (* creates a shelved goal *)
```

Comment:
It is not necessary to provide the argument to eexists here. Not sure if it is worth mentioning or abiding by. Typically in developments, if your goal will get automatically solved, its probably not worth naming it. There are of course exceptions to this. If you are solving the goal that will solve the named variable, sometimes it helps to provide the existential variable in the term, say given to `exact`.